### DISCUSSION TITLE
Where are Focus Groups stored?

### DISCUSSION THREAD

**Discussion by Morpheus0x** [quicken/filefocus#7] [General]
I really enjoy working with File Focus and think it provides the best features of any tab/file favoring extension. However, I am unable to find where the groups are stored by default. I do not want to use the global storage option, since I want project specific groups only. I am developing via [Remote SSH](https://github.com/jeanp413/open-remote-ssh) and the filefocus plugin is installed on the remote vscode server, if that makes any difference. The readme states:
> By default, focus groups are stored in your workspace allowing you to define a different set of focus groups per workspace.

But there are no new files in my workspace on the remote server, searching my local system(`grep -r my-favourited-file.php ~`) for the name of a file I have added to a group doesn't find any file containing that string...
Reading the code, the only reference to a file being read is the `.filefocus.json`, since I didn't create this and it isn't automatically created in my workspace I want to know where the default location is...

**Comment by quicken** [comment]
Have you checked if turning off "use global storage" in settings gets you the outcome you are after?

The extension uses the VSCode Data Storage API to store focus group information. Specifically, it uses either the global state or the workspace state of the API. The storage mode that is used is controlled in settings. 

Internally VSCode stores this information in a SQLite Database the location of which depends on your OS.
https://code.visualstudio.com/api/extension-capabilities/common-capabilities#data-storage

## Global State Storage
When using the global state focus groups are defined for all workspaces and synchronised across devices along with all other settings that typically get synchronised. 

An example use case for using global storage is if you have a desktop at home and a desktop in the office and you want to automatically sync focus groups between those devices; the global state will let you do that.

## Workspace State Storage
To define unique focus groups per workspace you can opt to instead store focus groups in the workspace state. This option would allow you to have different focus groups per workspace. 

**To switch to workspace storage open settings and uncheck the option "Use Global Storage".** 

Note that the settings themselves are also dependent on the scope. After switching storage modes you may have to restart the application and you may lose your current focus groups as the extension will now look at a different data store.

## ".filefocus.json" file
The ".filefocus.json" file is a project-specific file that you would need to manually create at the root of your project. The extension currently treats this file as "read-only" and does not create this file. 

You can create this file if you want to specifically create some static focus groups within a project. If a ".filefocus.json" file exists in a project those focus groups are shown along with the focus groups stored in data storage.  However, as pointed out only those groups in data storage are editable.

An example use case is making a focus group that brings together the various configuration and documentation files for a project.

I'm really happy to hear that the extension is getting some usage other than on my machine. :) I hope the above information helps if not please ask or outline your suggestions.

Cheers

Marcel
